src/Magnum/Math/Geometry/Intersection.h

```diff
@@ -167,25 +323,246 @@ template<class T> bool pointFrustum(const Vector3<T>& point, const Frustum<T>& f
+            return false;
+    }
+
+    return true;
+}
+
+
+template<class T> bool pointCone(const Vector3<T>& p, const Vector3<T>& origin, const Vector3<T>& normal, const Deg<T> angle) {
+    const T x = T(1)+Math::pow<T>(Math::tan<T>(angle/T(2)), T(2));
+
+    return pointCone(p, origin, normal, x);
+}
+
+template<class T> bool pointCone(const Vector3<T>& p, const Vector3<T>& origin, const Vector3<T>& normal, const T tanAngleSqPlusOne) {
+    const Vector3<T> c = p - origin;
+    const T lenA = dot(c, normal);
+
+    return lenA >= 0 && c.dot() <= lenA*lenA*tanAngleSqPlusOne;
+}
+
+template<class T> bool pointDoubleCone(const Vector3<T>& p, const Vector3<T>& origin, const Vector3<T>& normal, const Deg<T> angle) {
+    const T x = T(1)+Math::pow<T>(Math::tan<T>(angle/T(2)), T(2));
+
+    return pointDoubleCone(p, origin, normal, x);
+}
+
+template<class T> bool pointDoubleCone(const Vector3<T>& p, const Vector3<T>& origin, const Vector3<T>& normal, const T tanAngleSqPlusOne) {
+    const Vector3<T> c = p - origin;
+    const T lenA = dot(c, normal);
+
+    return c.dot() <= lenA*lenA*tanAngleSqPlusOne;
+}
+
+template<class T> bool sphereConeView(const Vector3<T>& sphereCenter, const T radius, const Matrix4<T>& coneView, const Rad<T> angle) {
+    const Float sinAngle = Math::sin(angle/T(2)); /* precomputable */
+    const Float cosAngle = Math::cos(angle/T(2)); /* precomputable */
+    const Float tanAngle = Math::tan<T>(angle/T(2));
+    const Float tanAngleSquaredPlusOne = T(1)+tanAngle*tanAngle;
+
+    return sphereConeView(sphereCenter, radius, coneView, sinAngle, cosAngle, tanAngle);
+}
+
+template<class T> bool sphereConeView(const Vector3<T>& sphereCenter, const T radius, const Matrix4<T>& coneView, const T sinAngle, const T cosAngle, const T tanAngle) {
+    /* Axis align cone */
+    const Vector3<T> center = coneView.transformPoint(sphereCenter);
+
+    /* Test against plane which determins whether to test against shiften cone or center-sphere */
+    if (center.z() > -radius*cosAngle) {
+        /* Axis aligned point - cone test shifted so that the cones surface is extended by `radius` */
+        const T coneRadius = tanAngle*(center.z() + radius/sinAngle);
+        return center.xy().dot() <= coneRadius*coneRadius;
+    }
+
+    return false;
+}
+
+template<class T> bool sphereCone(
+        const Vector3<T>& sCenter, const T radius,
+        const Vector3<T>& origin, const Vector3<T>& normal, const Rad<T> angle) {
+
+    const T sinAngle = Math::sin(angle / T(2)); /* precomputable */
+    const T tanAngleSquaredPlusOne = T(1)+Math::pow<T>(Math::tan<T>(angle/T(2)), T(2));
+
+    return sphereCone(sCenter, radius, origin, normal, sinAngle, tanAngleSquaredPlusOne);
+}
+
+template<class T> bool sphereCone(
+        const Vector3<T>& sCenter, const T radius,
+        const Vector3<T>& origin, const Vector3<T>& normal, const T sinAngle, const T tanAngleSquaredPlusOne) {
```

Comment:
Not sure about the indentation here. Maybe everything on one line? Or indent by 4 spaces and put the `{` on a new line?

Similarly above.

Reply:
> Or indent by 4 spaces and put the { on a new line?

I like that option 👍